### ISSUE TITLE
Restore loop in file-based programs

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#80242]
Repro: 

```cs
#:package Newtonsoft.Json:13.0.3
```

Expected: A diagnostic is reported for the bad `:` separator (`@` is expected) and restore fails
Actual: No diagnostic is shown in IDE, restore keeps happening in a loop

Found by @davidfowl
